src/check_result.jl

```diff
@@ -33,6 +33,12 @@ check_equal(x::Zero, y::Zero; kwargs...) = @test true
 check_equal(x::DoesNotExist, y::Nothing; kwargs...) = @test true
 check_equal(x::Nothing, y::DoesNotExist; kwargs...) = @test true
 
+check_equal(::ChainRulesCore.NotImplemented, x; kwargs...) = @test_broken false
+check_equal(x, ::ChainRulesCore.NotImplemented; kwargs...) = @test_broken false
```

Comment:
I wonder if throughout this section we should 
```suggestion
check_equal(x, y::ChainRulesCore.NotImplemented; kwargs...) = @test_broken x == y
```
Since seems like it gives a good message in testing outputs:

```
julia> @test_broken @not_implemented("derivative of foo(x,y) w.r.t y") == 1
Test Broken
  Expression: #= REPL[23]:1 =# @not_implemented("derivative of foo(x,y) w.r.t y") == 1
  ```

Reply:
I am a bit worried that `x == y` might evaluate to true sometimes in the case below where `x` and `y` are both instances of `NotImplemented`. But maybe your suggestion is to only use `x == y` in the other cases where the types are different?

Reply:
I didn't think about the case where they were both `NotImplemented`.
I wonder if that case should do `@test x == y`
I guess that case comes up when checking `add!!` ?
Not sure.

```julia
julia> @not_implemented("a") == @not_implemented("b")
false

julia> @not_implemented("a") == @not_implemented("a")
true
```

For other cases seems like better output is more good

Reply:
It's not completely clear to me if and how these non-constant tests could break but I didn't discover any problems with the more descriptive output in the tests, so I changed it.